### PR TITLE
Adding withCredentials support to mtz-auth-interceptor

### PR DESCRIPTION
**Pull request by admwx7**
Missed the PR a while back (life) but this shifts that functionality onto the existing element that's there to support all other auth options and deprecates the ill named cookie interceptor.

Also adds back in some tests that were removed for some reason, was able to test on FF / Edge / Chrome without issues.

Fixed some other minor issues such as a readme entry name being off and a missing element from the all-imports.

### CODE REVIEW

**Comment by admwx7** [comment]
@jay8t6 think you can take a look at this?  I went ahead and bumped the package.json to v0.6.0 as well in preparation just to be safe even though this should be fully backwards compatible.

Might want to look into adding a console.warn for anyone that loads in the now deprecated <mtz-cookie-interceptor> as well.

**Comment by jay8t6** [comment]
@admwx7 hey! Thanks for working on this! I’ll check it out in the morning.

File: mtz-auth-interceptor.html [hunk 1]
```diff
@@ -51,8 +53,12 @@
       intercept(event, eventName) {
         // Fallback for 1.x, use composedPath or rootTarget for 2.0
         const target = event.composedPath && (event.composedPath()[0] || Polymer.dom(event).rootTarget) || event.target;
-        if (target.getAttribute('with-auth') === '')
+        if (target.getAttribute('with-auth') === '') {
           this._applyAuth(event, eventName);
+        }
+        if (typeof this.withCredentials === 'boolean') {
```
Comment (jay8t6) [inline mtz-auth-interceptor.html:80]: @admwx7 do we need this check here since its always gonna pass?

**Review by jay8t6 — APPROVED** [review]

**Review by jay8t6 — DISMISSED** [review]

**Comment by jay8t6** [comment]
@admwx7 can you fix the conflict please?

**Review by jay8t6 — APPROVED** [review]
